homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -308,6 +324,92 @@ async def async_set_shuffle(self, shuffle):
                 "Service shuffle is not supported for non NetUSB sources."
             )
 
+    async def async_play_media(self, media_type: str, media_id: str, **kwargs) -> None:
+        """Play media."""
+        if self.state == STATE_OFF:
+            await self.async_turn_on()
+
+        if media_id:
+            parts = media_id.split(":")
+
+            if parts[0] == "list":
+                index = parts[3]
+
+                if index == "-1":
+                    index = "0"
+
+                await self.coordinator.musiccast.play_list_media(index, self._zone_id)
+                return
+
+            if parts[0] == "presets":
+                index = parts[1]
+                await self.coordinator.musiccast.recall_netusb_preset(
+                    self._zone_id, index
+                )
+                return
+
+            if parts[0] == "http":
+                await self.coordinator.musiccast.play_url_media(
+                    self._zone_id, media_id, "HomeAssistant"
+                )
+                return
+
+        raise HomeAssistantError(
+            "Only presets, media from media browser and http URLs are supported"
+        )
+
+    async def async_browse_media(self, media_content_type=None, media_content_id=None):
+        """Implement the websocket media browsing helper."""
+        if self.state == STATE_OFF:
+            raise HomeAssistantError(
+                "The device has to be turned on to be able to browse media."
+            )
+
+        if media_content_id:
+            media_content_path = media_content_id.split(":")
+            media_content_provider = await MusicCastMediaContent.browse_media(
+                self.coordinator.musiccast, self._zone_id, media_content_path, 24
+            )
+
+        else:
+            media_content_provider = MusicCastMediaContent(
```

Comment:
Looks like it would be enough to call the classmethod on the line below. I don't think we need the line here.

Reply:
oh damned, I totally missed that